### PR TITLE
When multiple NavigationModels are outside the module, only the first…

### PR DESCRIPTION
**Pull request by dlemures**
… one provided is processed.

### CODE REVIEW

File: dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java [hunk 1]
```diff
@@ -82,28 +82,33 @@ private void checkForParentsOutside(
...
+              compilerUtil.getPackageName(ancestorElement);
+          navigationModelBindingTarget.parentClass =
+              compilerUtil.getClassName(
+                  ancestorElement, navigationModelBindingTarget.parentPackage);
         }
       }
     }
   }
 
+  private TypeElement findClosestAncestorWithNavigationModelBinder(TypeElement element) {
```
Comment (stephanenicolas) [inline dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java]: The name of the method is confusing: it should return the closest ancestor, but it actually returns its model. I would suggest to split it into 2 methods.

File: dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java [hunk 2]
```diff
@@ -82,28 +82,33 @@ private void checkForParentsOutside(
       final NavigationModelBindingTarget navigationModelBindingTarget = target.getValue();
       // root inside module
       if (navigationModelBindingTarget.parentPackage == null) {
```
Comment (stephanenicolas) [inline dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java]: you should add a small comment here to say we use this field as caching.

File: dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java (hunk 1 above)
Comment (stephanenicolas) [reply dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java]: Or you could rename to `findClosestAncestorNavigationModelBinder`

File: dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java [hunk 3]
```diff
@@ -82,28 +82,33 @@ private void checkForParentsOutside(
...
-          // ancestor contains a NavigationModel field
-          if (getNavigationModelBinder(ancestorElement) != null) {
-            navigationModelBindingTarget.parentPackage =
-                compilerUtil.getPackageName(ancestorElement);
-            navigationModelBindingTarget.parentClass =
-                compilerUtil.getClassName(
-                    ancestorElement, navigationModelBindingTarget.parentPackage);
-            return;
-          }
+        TypeElement ancestorElement = findClosestAncestorWithNavigationModelBinder(element);
```
Comment (stephanenicolas) [inline dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java]: `ancestorNavigationModelBinderTypeElement`

**Comment by stephanenicolas** [comment]
great investigative job !

File: dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java (hunk 1 above)
Comment (stephanenicolas) [reply dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java]: forget it

File: dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java (hunk 2 above)
Comment (dlemures) [reply dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java]: added a comment explaining a little bit more how it works

File: dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java (hunk 3 above)
Comment (dlemures) [reply dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java]: after discussing offline, it is the ancestor and not it model

**Pull request merged**